flang/lib/Lower/Allocatable.cpp

```diff
@@ -137,25 +138,25 @@ static void genRuntimeInitCharacter(fir::FirOpBuilder &builder,
                                     mlir::Location loc,
                                     const fir::MutableBoxValue &box,
                                     mlir::Value len) {
-  auto callee =
+  mlir::FuncOp callee =
       box.isPointer()
           ? fir::runtime::getRuntimeFunc<mkRTKey(PointerNullifyCharacter)>(
                 loc, builder)
           : fir::runtime::getRuntimeFunc<mkRTKey(AllocatableInitCharacter)>(
                 loc, builder);
-  auto inputTypes = callee.getType().getInputs();
+  llvm::ArrayRef<mlir::Type> inputTypes = callee.getType().getInputs();
   if (inputTypes.size() != 5)
     fir::emitFatalError(
         loc, "AllocatableInitCharacter runtime interface not as expected");
   llvm::SmallVector<mlir::Value> args;
   args.push_back(builder.createConvert(loc, inputTypes[0], box.getAddr()));
   args.push_back(builder.createConvert(loc, inputTypes[1], len));
-  auto kind = box.getEleTy().cast<fir::CharacterType>().getFKind();
+  int kind = box.getEleTy().cast<fir::CharacterType>().getFKind();
```

Comment:
This is a silent type conversion. getFKind returns a `KindTy`. And a `KindTy` is actually `unsigned int`.

Probably stick with `Fortran::lower::KindTy`.

Reply:
Braced inits would have caught that for me, grr.